From a3e084d254947836ca8233dc543ccc7fef8e83e7 Mon Sep 17 00:00:00 2001
From: Rob Evers <revers@redhat.com>
Date: Mon, 23 Jun 2014 21:41:43 -0400
Subject: [scsi] lpfc 8.3.45: Added dport mailbox pass through support

Message-id: <1403559726-20488-2-git-send-email-revers@redhat.com>
Patchwork-id: 84433
O-Subject: [RHEL7.1 e-stor PATCH 01/24] scsi: lpfc 8.3.45: Added dport mailbox pass through support.
Bugzilla: 1088574
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: James Smart <james.smart@emulex.com>

Signed-off-by: James Smart <james.smart@emulex.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 2a2719d370b2ec31451462f2775986e42cbeb39f)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/lpfc/lpfc_bsg.c b/drivers/scsi/lpfc/lpfc_bsg.c
index 82134d2..ca2f4ea 100644
--- a/drivers/scsi/lpfc/lpfc_bsg.c
+++ b/drivers/scsi/lpfc/lpfc_bsg.c
@@ -4153,6 +4153,7 @@ lpfc_bsg_handle_sli_cfg_mbox(struct lpfc_hba *phba, struct fc_bsg_job *job,
   if (subsys == SLI_CONFIG_SUBSYS_FCOE) {
    switch (opcode) {
    case FCOE_OPCODE_READ_FCF:
+   case FCOE_OPCODE_GET_DPORT_RESULTS:
     lpfc_printf_log(phba, KERN_INFO, LOG_LIBDFC,
       "2957 Handled SLI_CONFIG "
       "subsys_fcoe, opcode:x%x\n",
@@ -4161,6 +4162,8 @@ lpfc_bsg_handle_sli_cfg_mbox(struct lpfc_hba *phba, struct fc_bsg_job *job,
        nemb_mse, dmabuf);
     break;
    case FCOE_OPCODE_ADD_FCF:
+   case FCOE_OPCODE_SET_DPORT_MODE:
+   case LPFC_MBOX_OPCODE_FCOE_LINK_DIAG_STATE:
     lpfc_printf_log(phba, KERN_INFO, LOG_LIBDFC,
       "2958 Handled SLI_CONFIG "
       "subsys_fcoe, opcode:x%x\n",
diff --git a/drivers/scsi/lpfc/lpfc_bsg.h b/drivers/scsi/lpfc/lpfc_bsg.h
index 67f7d0a..a94d4c9 100644
--- a/drivers/scsi/lpfc/lpfc_bsg.h
+++ b/drivers/scsi/lpfc/lpfc_bsg.h
@@ -231,6 +231,8 @@ struct lpfc_sli_config_emb0_subsys {
 #define SLI_CONFIG_SUBSYS_FCOE  0x0C
 #define FCOE_OPCODE_READ_FCF  0x08
 #define FCOE_OPCODE_ADD_FCF  0x09
+#define FCOE_OPCODE_SET_DPORT_MODE 0x27
+#define FCOE_OPCODE_GET_DPORT_RESULTS 0x28
 };
 
 struct lpfc_sli_config_emb1_subsys {
-- 
1.7.1